socMapping/StockSOC_ProcessPoints.py

```diff
@@ -0,0 +1,230 @@
+min_feat=2
+max_feat=3
+
+
+# In[6]:
+
+
+### Process stock. To process SOC change to False ###
+processStock = True
+
+
+# In[7]:
+
+
+# Function to calcualte normalized difference vegetation index
+def calcNDVI(red, nir):
+    return pd.DataFrame((nir - red)/(nir + red))
+
+
+# In[8]:
+
+
+# Function to calcualte soil-adjusted total vegetation index
+def calcSATVI(red, swir1, swir2):
+    return pd.DataFrame(((swir1 -red)/(swir1 + red+0.5)) * 1.5 - (swir2/2))
+
+
+# In[9]:
+
+
+# Function to calcualte normalized burn ratio 2
+def calcNBR2(swir1, swir2):
+    return pd.DataFrame((swir1 -swir2)/(swir1 + swir2))
+
+
+# In[10]:
+
+
+# Function to calcualte soil organic carbon index
+def calcSOCI(blue, green, red):
+    return pd.DataFrame(blue/(red * green))
+
+
+# In[11]:
+
+
+# Function to calcualte bare soil index
+def calcBSI(blue, red, nir, swir1):
+    return pd.DataFrame((swir1 + red) -(nir + blue) / (swir1 + red) + (nir + blue))
+
+
+# In[12]:
+
+
+# Function to calcualte adjusted R2
+def adjust_r2(r2, num_examples, num_features):
+    coef = (num_examples - 1) / (num_examples - num_features - 1) 
+    return 1 - (1 - r2) * coef
+
+
+# In[13]:
+
+
+# Open the tabular data that was output from StockSOC_ExtractPoints
+with open(inPickle, 'rb') as f:
+    pointsDFs = pickle.load(f)
+
+
+# In[14]:
+
+
+# Initialize for tabular data to be output to CSV
+regResults = pd.DataFrame(columns = ['Date', 'R2', 'Adjusted_R2', 'RMSE', 'BestFeatures'])
+
+
+# In[15]:
+
+
+# Iterate through the dictionary of tabular one date at a time to find the set of variables
+# that gievs the lowest R2 value
+for iteration, key in enumerate(pointsDFs):
+    points = pointsDFs[key]
+    if (points['B3'].isna().sum() / len(points.index) < 0.2): # If under 20% of the values are NA (cloud masked)
+        print("Processing " + key + ": " + (str(len(pointsDFs.keys())-iteration-1)) + 
+              " images to go      ", end = "\r")
+        points.dropna(inplace=True)
+        points['ndvi']= calcNDVI(points['B4'].astype(float), points['B8'].astype(float))
+        points['satvi']= calcSATVI(points['B4'].astype(float), points['B11'].astype(float), 
+                                   points['B12'].astype(float))
+        points['nbr2']= calcNBR2(points['B11'].astype(float), points['B12'].astype(float))
+        points['soci']= calcSOCI(points['B2'].astype(float), points['B3'].astype(float), 
+                                 points['B4'].astype(float))
+        points['bsi']= calcBSI(points['B2'].astype(float), points['B4'].astype(float), 
+                                 points['B8'].astype(float), points['B11'].astype(float))
+
+        x = pd.DataFrame(points.drop([SOC, BD, PointLabel, 'stock'], axis=1))
+        if (processStock):
+            y = points[['stock']]
+        else:
+            y = points[[SOC]]
+        # Set up for leave one out processig
+        loo = LeaveOneOut()
```

Comment:
I'm pretty sure this isn't being used as you define it below in `cv = LeaveOneOut()` 
```suggestion
```
```suggestion
        loo = LeaveOneOut()
```